### PR TITLE
Expo/HomeScreen Integration tests

### PR DESCRIPTION
**Pull request by konti-aditi**
Date: 28-05-25

Developer Name: Aditi Shinde

---

## Issue Ticket Number
#520 

## Description
This PR contain integration tests for HomeScreen 

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

test1

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2025-05-30 210527

</details>

### CODE REVIEW

File: __tests__/HomeScreen.test.tsx [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+  it("submits OOO form and updates status", async () => {
+    mockSubmitOOOForm.mockResolvedValueOnce(true);
+    const { getByText, getByTestId } = render(<HomeScreen />);
+    fireEvent.press(getByText("Submit OOO"));
+    fireEvent.press(getByTestId("mock-form-submit"));
+    await waitFor(() => {
+      expect(mockSubmitOOOForm).toHaveBeenCalled();
+      expect(mockFetchUserStatus).toHaveBeenCalled();
+    });
```
Comment (VinuB-Dev) [inline __tests__/HomeScreen.test.tsx:121]: can we add the success alert assertions also here just like we did for error in lines 134 - 137?

File: __tests__/HomeScreen.test.tsx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      submitOOOForm: mockSubmitOOOForm,
+      cancelOOO: mockCancelOOO,
+      loading: false,
+    });
+    mockCancelOOO.mockResolvedValueOnce(true);
+    const { getByText } = render(<HomeScreen />);
+    fireEvent.press(getByText("Cancel OOO"));
+    await waitFor(() => {
+      expect(mockCancelOOO).toHaveBeenCalled();
+      expect(mockFetchUserStatus).toHaveBeenCalled();
```
Comment (VinuB-Dev) [inline __tests__/HomeScreen.test.tsx:173]: can we add the success alert assertions when cancel happens successfully?

File: __tests__/HomeScreen.test.tsx [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+      userStatus: { data: { currentStatus: { state: "OOO" } } },
+      submitOOOForm: mockSubmitOOOForm,
+      cancelOOO: mockCancelOOO,
+      loading: false,
+    });
+    mockCancelOOO.mockRejectedValueOnce(new Error("Cancel failed"));
+    const { getByText } = render(<HomeScreen />);
+    fireEvent.press(getByText("Cancel OOO"));
+    await waitFor(() => {
+      expect(mockCancelOOO).toHaveBeenCalled();
```
Comment (VinuB-Dev) [inline __tests__/HomeScreen.test.tsx:189]: Let's also check through error message alert assertion on whether the catch part of code was called here.

File: __tests__/utils.test.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+  it("formats valid timestamp (s)", () => {
+    const dateStr = formatDateTime(1704067200); // Jan 1, 2024 in seconds
+    expect(dateStr).toBe("January 1, 2024");
+  });
+
+  it("returns 'Invalid date' for falsy input", () => {
+    expect(formatDateTime("")).toBe("Invalid date");
+  });
+});
```
Comment (VinuB-Dev) [inline __tests__/utils.test.tsx:34]: can we also have test cases which test for undefined, null and NaN here since these might popup when we get data from backend and we can make sure our function works as expected in these scenarios.

**Comment by VinuB-Dev** [comment]
@Aditys018 The coverage report for HomeScreen flags lines 28-61 as having some uncovered parts (branch coverage is 86%). I know handleCancelOOO is in that area and has tests but maybe we missed some branches, can you please check.
We can utilise HTML coverage report to check which branch was missed, just add html to `coverageReporters` in `jest.config.js`, it'll generate the html coverage report when we run the test suite.
